File: Source/Camping Mod 2.0/common/rikmuld/camping/inventory/slot/SlotState.java
package rikmuld.camping.inventory.slot;

import net.minecraft.inventory.IInventory;

public class SlotState extends SlotDisable {

	int stateX;
	int stateY;

	public SlotState(IInventory inv, int id, int x, int y)
	{
		super(inv, id, x, y);
		stateX = xFlag;
		stateY = yFlag;
	}

	@Override
	public void enable()
	{
		xDisplayPosition = stateX;
		yDisplayPosition = stateY;
	}

	public void setStateX(int state)
	{
		stateX = xFlag - (18 * state);
	}

	public void setStateY(int state)
	{
		stateY = yFlag - (18 * state);
	}
}

File: scala/com/rikmuld/camping/common/inventory/SlotState.scala
package com.rikmuld.camping.common.inventory

import com.rikmuld.corerm.gui.slots.SlotDisable
import net.minecraft.inventory.{IInventory, Slot}

class SlotState(inv: IInventory, id: Int, x: Int, y: Int) extends Slot(inv, id, x, y) with SlotDisable {
  var stateX: Int = xFlag
  var stateY: Int = yFlag

  override def enable() {
    xPos = stateX
    yPos = stateY
  }

  def setStateX(state: Int) =
    stateX = xFlag - (18 * state)

  def setStateY(state: Int) =
    stateY = yFlag - (18 * state)
}